xds/src/main/java/io/grpc/xds/XdsLoadBalancer2.java

```diff
@@ -245,4 +248,18 @@ public void updateBalancingState(ConnectivityState newState, SubchannelPicker ne
   interface LookasideLbFactory {
     LoadBalancer newLoadBalancer(Helper helper, AdsStreamCallback adsCallback);
   }
+
+  private static final class LookasideLbFactoryImpl implements LookasideLbFactory {
```

Comment:
You don't need to create a separate `LookasideLbFactory` interface. `LoadBalancer.Factory` suffices. You can make this class non-static and access `AdsCallback` instance directly. You would need a secondary `XdsLoadBalancer2` constructor that takes an `AdsStreamCallback` instance. Make your implementation for `AdsStreamCallback` a standalone class with name (instead of anonymous), and make it package-private with `@VisibleForTesting`. You can access it in test. I feel like this is more elegant.

Reply:
Technically this is verify difficult:

```
XdsLoadBalancer2(Helper helper) {
    this(helper, new LookasideLbFactory(), new FallbackLbFactory());
  }
```

won't compile because can not referece non-static `LookasideLbFactory` in constructor.

Also, I'm not a fan of having `LookasideLbFactory` appear like depending the state of `XdsLoadBalancer2` and appear like it can be changed while the state of `XdsLoadBalancer2` changes.

Reply:
Oh, I forget that won't compile...

But still, you can pass your `AdsStreamCallback` instance into the constructor of `LookasideLbFactory`, which extends `LoadBalancer.Factory`. We should try to minimize the number of newly interfaces.

Reply:
```
XdsLoadBalancer2(Helper helper) {
    this(helper, new LookasideLbFactory(adsCallback), new FallbackLbFactory());
  }
```

This or other approaches also won't compile, the technical difficulty is not trivial.

Reply:
If an object you are injecting to the constructor depends on the class implementation, then you should not _only_ use the mocked object in test, because that object depends on implementation.

Reply:
No, I mean you can make `LookasideLbFactory` static, not depending on `XdsLoadBalancer2`'s state. You just need a `LookasideLbFactory(AdsStreamCallback)` constructor.

Reply:
I knew what you mean. Look at my second snippet, it won't compile.

Reply:
The adsStreamCallback passed in the constructor depends on and changes the state of XdsLoadBalancer2.